File: python/src/deployment/order_service.py
from src.cloud.aws.s3 import create_s3_bucket
from src.deployment.service import Service

import pulumi


class OrderService(Service):
    def __init__(self, config: pulumi.Config):
        super().__init__(config)

    def deploy(self):
        # Deployment logic for Order Service. Just an example.
        bucket = create_s3_bucket(f"{self.service_name}-bucket", force_destroy=True)
        pulumi.export("order_service_bucket", bucket._name)

File: typescript/src/deployment/order-service.ts
import { Input } from "@pulumi/pulumi";
import { createEC2Instance } from "../cloud/aws/ec2";
import { Service } from "./service";

export class OrderService extends Service {
    async deploy(): Promise<{ [key: string]: Input<string>; }> {
        const ec2Instance = await createEC2Instance("order-service-instance", "t2.micro", "ami-0c55b159cbfafe1f0");
        return {
            ec2InstanceId: ec2Instance.id
        };
    }
}